Content.Server/Animals/Systems/EggLayerSystem.cs

```diff
@@ -77,7 +80,7 @@ public bool TryLayEgg(EntityUid uid, EggLayerComponent? component)
                 return false;
             }
 
-            hunger.CurrentHunger -= component.HungerUsage;
+            _hunger.ModifyHunger(uid, hunger.CurrentHunger - component.HungerUsage, hunger);
```

Comment:
Should this be SetHunger? I think this will add the difference which doesn't seem intended.